packages/lesswrong/lib/collections/tags/helpers.ts

```diff
@@ -93,3 +94,33 @@ export const userIsSubforumModerator = (user: DbUser|UsersCurrent|null, tag: DbT
   if (!user || !tag) return false;
   return tag.subforumModeratorIds?.includes(user._id);
 }
+
+/**
+ * Sort tags in order of: core-ness, score, name (alphabetical). If we don't have the scores, sort only by core-ness.
+ */
+export function stableSortTags<
+  T extends {name: string; core: boolean},
+  TR extends {baseScore: number} | null | undefined
+>(tagInfo: Array<{ tag: T; tagRel: TR }>): Array<{ tag: T; tagRel: TR }> {
+  return [...tagInfo].sort((a, b) => {
+    const tagA = a.tag;
+    const tagB = b.tag;
+    const tagRelA = a.tagRel;
+    const tagRelB = b.tagRel;
+
+    if (tagA.core !== tagB.core) {
+      // Core tags come first with isFriendlyUI, last otherwise
+      return (tagA.core ? -1 : 1) * (isFriendlyUI ? 1 : -1);
+    }
+
+    if (tagRelA && tagRelB) {
+      if (tagRelA.baseScore !== tagRelB.baseScore) {
+        return (tagRelB.baseScore || 0) - (tagRelA.baseScore || 0);
+      }
+
+      return tagA.name.localeCompare(tagB.name);
```

Comment:
The idea is that if you have the tags from the backend which were sorted by tag relevance, but now you don't have the tagrel score on the frontend, you should leave the order as is after sorting by `core` so as not to mess up the existing ordering.

The way the PR ended up this doesn't actually make any difference, but if e.g. stableSortTags were applied to `post.tags` here it would make a difference:
```typescript
const sortedTagInfo = results
    ? stableSortTags(results.filter((tagRel) => !!tagRel?.tag).map((tr) => ({ tag: tr.tag!, tagRel: tr })))
    : post.tags.map((tag) => ({ tag, tagRel: undefined }));
 ```